examples/calculator.rs

```diff
@@ -104,9 +110,9 @@ fn update_button_border<W: Widget>(
     state: WidgetState,
 ) {
     match state.state() {
-        Button::STATE_HOVERED => widget.border_color(BinaryColor::On),
-        Button::STATE_PRESSED => widget.border_color(BinaryColor::Off),
-        _ => widget.border_color(BinaryColor::Off),
+        Button::STATE_HOVERED => widget.border_properties.border_color(BinaryColor::On),
```

Comment:
It's not ideal users have to refer to the property field here. Maybe name the constructor methods `with_` or use builder objects?